src/v2/view-builder/views/email/BaseAuthenticatorEmailView.js

```diff
@@ -77,20 +77,29 @@ const Body = BaseFormWithPolling.extend(Object.assign(
       BaseForm.prototype.triggerAfterError.apply(this, arguments);
       this.stopPolling();
 
-      // Wait 1 min before polling again & do not show error message when encounter rate limit error
-      if (error.responseJSON?.errorSummaryKeys?.includes('tooManyRequests') ||
-        error.responseJSON?.errorCode === 'E0000047') {
+      // Polling needs to be stopped if it's a form error and session is invalid
+      if (error.responseJSON?.errorSummaryKeys?.includes('idx.session.expired')) {
+        return;
+      }
+
+      // Wait 1 min before polling again & do not show error message when polling encounter rate limit error
+      if (error.responseJSON?.errorSummaryKeys?.includes('tooManyRequests')
+          || error.responseJSON?.errorCode === 'E0000047') {
+
+        // if there is intent (ie. user submit OTP), resume polling
+        if (error.responseJSON?.errorIntent) {
+          this.startPolling(this.options.appState.get('dynamicRefreshInterval'));
+          return;
+        }
+
         setTimeout(() => {
           model.trigger('clearFormError');
         }, 0);
         setTimeout(() => {
           this.startPolling(this.options.appState.get('dynamicRefreshInterval'));
         }, 60000);
-        return;
-      }
-
-      // Polling needs to be resumed if it's a form error and session is still valid
-      if(!error.responseJSON?.errorSummaryKeys?.includes('idx.session.expired')) {
+      } else {
+        // Resume polling for other error
```

Comment:
what does this message means `Resume polling for other error` ??

Reply:
> what does this message means `Resume polling for other error` ??

IIUC this is essentially preserving [the logic from PR 1642](https://github.com/okta/okta-signin-widget/pull/1642/files#diff-17fe5ab3798226da710a0498ac513740370f100c865cfbb0c4e022cabab04be0), i.e. if we got `idx.session.expired` then handle it one way and for any other sort of error handle it another way?

Reply:
Also we need to keep polling since `saveForm()` also `stopPolling()`  and if `saveForm()` encounter error (such as invalid OTP), we need to resume the polling.

Reply:
we don't need this comment anymore.. we can remove it.

Reply:
or we can say.. ```If there are any other valid non blocking errors then continue polling```

Reply:
sure let me add it that comment instead